### DISCUSSION TITLE
Deployment automation/scripting for AWS and/or Cloudflare

### DISCUSSION THREAD

**Discussion by PhilipMathieu** [protomaps/PMTiles#324] [Q&A]
I'm developing a guide to deploying webmap projects that will be using PMTiles. The intended audience is graduate students in a CS/DS program that produce analyses resulting in a geospatial dataset, typically geojson, and want to deploy a website showing their work. The intent is to minimize the amount of additional knowledge required - i.e., even though some of them will want to learn AWS for their careers, many will not and I'd like to make it as simple as possible. **Is there any reason not to develop a deployment script that automates the steps described at https://docs.protomaps.com/deploy/aws?** Something that would run the `npm` commands, use the CLI to configure services, etc., with appropriate prompts to the user to make sure they don't do something wrong.

(Full disclosure - I'm still pretty new to cloud services myself.)

**Comment by bdon** [comment]
To directly answer your question - yes, deployment automation is useful for AWS. @daniel-j-h has open sourced an example using AWS CDK here: https://github.com/robofarmio/aws-cdk-protomaps but some others prefer Terraform/OpenTofu; we'd like to eventually have these in the docs but right now the steps are to address the widest possible audience who may not use CDK/TF

to indirectly answer your question - for a student project using the AWS deployment option seems like overkill, and that would tie their project to AWS instead of being platform-agnostic; I would suggest ignoring anything in the `Accelerating PMTiles` part of the docs as that is really best suited for high traffic web properties. I would recommend they learn the `rclone` tool to upload PMTiles to any platform (Cloudflare R2 and DigitalOcean Spaces are good choices) and then consume PMTiles directly in the browser.

Finally if their end result is viable to fit in a GeoJSON, that is a simpler option and you can avoid using PMTiles at all.
